src/main/java/com/uber/cadence/internal/replay/MutableMarkerHandler.java

```diff
@@ -0,0 +1,163 @@
+/*
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package com.uber.cadence.internal.replay;
+
+import com.uber.cadence.EventType;
+import com.uber.cadence.HistoryEvent;
+import com.uber.cadence.MarkerRecordedEventAttributes;
+import com.uber.cadence.converter.DataConverter;
+import com.uber.cadence.workflow.Functions.Func1;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+class MutableMarkerHandler {
```

Comment:
Rename as it is used not only for MutableSideEffectMarker

Reply:
Done.